packages/expo/src/launch/registerRootComponent.tsx

```diff
@@ -26,33 +24,22 @@ export default function registerRootComponent<P extends InitialProps>(
     qualifiedComponent = withDevTools(component);
   }
 
-  if (Platform.OS !== 'web') {
-    AppRegistry.registerComponent('main', () => qualifiedComponent);
-  } else if (
+  AppRegistry.registerComponent('main', () => qualifiedComponent);
+  if (
+    Platform.OS === 'web' &&
     // Skip querying the DOM if we're in a Node.js environment.
     typeof document !== 'undefined'
   ) {
-    let tag = document.getElementById('root');
-
-    if (!tag) {
-      tag = document.getElementById('main');
-      if (process.env.NODE_ENV !== 'production') {
-        // This block will be removed in production
-        if (tag) {
-          console.warn(
-            'Mounting the root React component to an HTML element with id "main" is deprecated. Use id "root" instead.'
-          );
-        }
+    AppRegistry.registerComponent('main', () => qualifiedComponent);
+    const rootTag = document.getElementById('root');
+    if (process.env.NODE_ENV !== 'production') {
+      if (!rootTag) {
+        throw new Error('Required HTML element with id "root" was not found in the document HTML.');
       }
     }
-
-    if (!tag) {
-      throw new Error(
-        'Required HTML element with id "root" was not found in the document HTML. This is required for mounting the root React component.'
-      );
-    }
-
-    const rootTag = createRoot(tag);
-    rootTag.render(React.createElement(qualifiedComponent));
+    AppRegistry.runApplication('main', {
+      rootTag,
```

Comment:
Is this property name something we have control over? rootElement would be a more accurate name than rootTag.

Reply:
This is the property name